enhancements/cluster-logging/cluster-logging-otel-support.md

```diff
@@ -0,0 +1,294 @@
+* Modify the vector log collector deployment to write to **LokiStack** over OTLP and additionally include select ViaQ attributes
+* Update the **loki-operator** to [provide an API](https://issues.redhat.com/browse/LOG-5673) to configure Loki to ingest logs through an OTLP endpoint
+* Update the **loki-operator** to provide sane defaults to the new LokiStack OTEL API for server-side indexing of OTEL attributes to improve query performance
+* Modify the **logging-view-plugin** to use OTEL attributes that correspond to ViaQ *log_type*, *log_source*, *kubernetes.container_name*, *kubernetes.namespace_name*, *kubernetes.pod_name*
+
+### Workflow Description
+
+The following workflow describes deployment of a full logging stack to collect and forward logs to a Red Hat managed log store, **LokiStack**, using OTLP.
+
+**cluster administrator** is a human responsible for:
+
+* Managing and deploying day 2 operators
+* Managing and deploying an in-cluster LokiStack
+* Managing and deploying a cluster-wide log forwarder
+
+**cluster-observability-operator** is an operator responsible for:
+
+* managing and deploying observability operands and console plugins (e.g logging-view-plugin)
+
+**loki-operator** is an operator responsible for managing a LokiStack instance.
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding.
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **cluster-observability-operator**
+1. Deploys the Red Hat **loki-operator** which supports server-side indexing and OTLP ingestion endoint
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+    1. **In Tech Preview, additionally annotated to instruct the operator to use the OTEL datamodel in lieu of ViaQ**
+
+The **cluster-observability-operator**:
+
+1. Deploys the console-logging-plugin for reading logs in the OpenShift console
+
+The **loki-operator**
+
+1. Deploys the **LokiStack** for storing logs in-cluster
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+
+### API Extensions
+
+During the time this feature is considered Tech Preview, instances of **ClusterLogForwarder** will be annotated to indicate to the
+operator that logs will be forwarded to **LokiStack** using OTLP:
+
+```
+   observability.openshift.io/enable-lokistack-otlp: [[LOKISTACK_NAMES]]
+```
+* LOKISTACK_NAMES := Comma delimited list of **clusterlogforwarder.spec.output[].name** of type **lokiStack**
+
+Example of configuring multiple outputs:
+
+```yaml
+apiVersion: "observability.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  annotations:
+    logstore.observability.openshift.io/otlp: my-loki-otel,my-other-loki-otel
```

Comment:
> configuration attribute on the output

What does this look like?  The intent of keeping an annotation is to not have to add a field into the API spec which because obsolete once writing to LokiStack only via OTLP

>  once the feature becomes GA but customers still need a "mixed output" they can continue to use the attribute and we just switch the meaning of "no value provided" at some point in the future

I anticipate adding a complimentary annotation when the feature goes GA to allow enablement of ViaQ for lokiStack which goes away GA+1 since that is when the UI is expected to reach parity

Reply:
Discussion we resolved:
* Rely only upon TP enablement annotation
* Add field to lokiStack output: `dataModel=(""|viaq|otel)`
* TP: `""=viaq`
* GA: `""=otel`

cc @Clee2691